python/ray/train/base_trainer.py

```diff
@@ -767,13 +767,16 @@ def _reconcile_scaling_config_with_trial_resources(
                     scaling_config = trainer_cls._validate_scaling_config(
                         scaling_config
                     )
+                else:
+                    return ScalingConfig()
```

Comment:
`scaling_config` is None when we create a trainer without specifying `scaling_config`. We should not build scaling config from `trial_resources`, but return a default `ScalingConfig`.